src/main/java/uk/gov/pay/directdebit/webhook/gocardless/services/handlers/SendEmailsForGoCardlessEventsHandler.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.pay.directdebit.webhook.gocardless.services.handlers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoMandateIdException;
+import uk.gov.pay.directdebit.events.exception.GoCardlessEventHasNoPaymentIdException;
+import uk.gov.pay.directdebit.events.model.GoCardlessEvent;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.mandate.services.MandateQueryService;
+import uk.gov.pay.directdebit.notifications.services.UserNotificationService;
+import uk.gov.pay.directdebit.payments.exception.PaymentNotFoundException;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentId;
+import uk.gov.pay.directdebit.payments.model.Payment;
+import uk.gov.pay.directdebit.payments.services.PaymentQueryService;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.MANDATES;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYMENTS;
+import static uk.gov.pay.directdebit.events.model.GoCardlessResourceType.PAYOUTS;
+
+public class SendEmailsForGoCardlessEventsHandler {
+
+    private static Predicate<GoCardlessEvent> byValidEventsForMandateResourceTypes =
+            event -> MANDATES.equals(event.getResourceType());
+    
+    private static Predicate<GoCardlessEvent> byValidEventsForPaymentsAndPayoutsResourceTypes =
+            event -> Set.of(PAYMENTS, PAYOUTS).contains(event.getResourceType());
+
+    private final MandateQueryService mandateQueryService;
+    private final UserNotificationService userNotificationService;
+    private final PaymentQueryService paymentQueryService;
+
+    @Inject
+    public SendEmailsForGoCardlessEventsHandler(MandateQueryService mandateQueryService,
+                                                UserNotificationService userNotificationService,
+                                                PaymentQueryService paymentQueryService) {
+        this.mandateQueryService = mandateQueryService;
+        this.userNotificationService = userNotificationService;
+        this.paymentQueryService = paymentQueryService;
+    }
+
+    public void sendEmails(List<GoCardlessEvent> events) {
+        events.stream().filter(byValidEventsForMandateResourceTypes).forEach(e -> sendEmailForMandateEvents(e));
+        events.stream().filter(byValidEventsForPaymentsAndPayoutsResourceTypes).forEach(e -> sendEmailForPaymentEvents(e));
+    }
+
+    private void sendEmailForPaymentEvents(GoCardlessEvent event) {
+        GoCardlessPaymentId goCardlessPaymentId = event.getLinksPayment()
+                .orElseThrow(() -> new GoCardlessEventHasNoPaymentIdException(event.getGoCardlessEventId()));
```

Comment:
If this was done after we know we have to send an email (similar to mandates with its `getMandate()` method), it would be a bit safer (less chance of an event we don’t care about for email purposes blowing us up) and increase consistency.